framework/Optimizers/SimulatedAnnealing.py

```diff
@@ -0,0 +1,718 @@
+      return set(new)
+
+  def _rejectOptPoint(self, traj, info, old):
+    """
+      Having rejected the suggested opt point, take actions so we can move forward
+      @ In, traj, int, identifier
+      @ In, info, dict, meta information about the opt point
+      @ In, old, dict, previous optimal point (to resubmit)
+    """
+    self._cancelAssociatedJobs(info['traj'], step=info['step'])
+    ## what do do if a point is rejected?
+    # TODO user option to EITHER rerun opt point OR cut step!
+    # initialize a new step
+    self._initializeStep(traj)
+    #self._acceptRerun[traj] = True #TODO: What if the model is inherintly stochastic ?!
+  # END resolving potential opt points
+  # * * * * * * * * * * * * * * * *
+
+  def _applyFunctionalConstraints(self, suggested, previous):
+    """
+      fixes functional constraints of variables in "point" -> DENORMED point expected!
+      @ In, suggested, dict, potential point to apply constraints to
+      @ In, previous, dict, previous opt point in consideration
+      @ Out, point, dict, adjusted variables
+      @ Out, modded, bool, whether point was modified or not
+    """
+    # assume no modifications until proved otherwise
+    modded = False
+    # are we violating functional constraints?
+    passFuncs = self._checkFunctionalConstraints(self.denormalizeData(suggested))
+    # while in violation of constraints ...
+    tries = 500
+    while not passFuncs:
+      modded = True
+      #  try to find new acceptable point
+      denormed = self.denormalizeData(suggested)
+      suggested, _= self._fixFuncConstraintViolations(suggested, previous) # need info, traj
+      denormed = self.denormalizeData(suggested)
+      self.raiseADebug(' ... suggested new opt {}'.format(denormed))
+      passFuncs = self._checkFunctionalConstraints(denormed)
+      tries -= 1
+      if tries == 0:
+        self.raiseAnError(NotImplementedError, 'No acceptable point findable! Now what?')
+    return suggested, modded
+  ###########
+  # Utility Methods #
+  ###########
+  def _temperature(self, fraction):
+    #return max(0.01,min(1,1-fraction))
+    return max(0.01,1-fraction)
+
+  def _coolingSchedule(self, iter, T0):
+    """TODO write the math here"""
+    #defaults
+    if T0 == None:
+      T0 = 1e4
+
+    if not self._coolingMethod:
+      self._coolingMethod = 'exponential'
+
+    if not self._CoolingParameters:
+      self._CoolingParameters['alpha'] = 0.94
+      self._CoolingParameters['beta'] = 0.1
+      self._CoolingParameters['c'] = 1.0
+      self._CoolingParameters['d'] = 1.0
+
+    type = self._coolingMethod
+    if type == 'linear':
+      beta = self._CoolingParameters['beta']
+      return T0 - iter * beta
+    elif type in ['exponential','geometric']:
+      alpha = self._CoolingParameters['alpha']
+      return alpha ** iter * T0
+    elif type == 'boltzmann':
+      d = self._CoolingParameters['d']
+      return T0/(np.log10(iter + d))
+    elif type == 'fast':
```

Comment:
you commented this out from the input reading...=> it has to be removed from here...